pkg/component/types.go

```diff
@@ -48,22 +48,8 @@ type ComponentStatus struct {
 type CombinedComponentList struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ListMeta   `json:"metadata,omitempty"`
-	S2IComponents     []Component        `json:"s2iComponents"`
-	DevfileComponents []DevfileComponent `json:"devfileComponents"`
-}
-
-type DevfileComponent struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              DevfileComponentSpec `json:"spec,omitempty"`
-	Status            ComponentStatus      `json:"status,omitempty"`
-}
-
-type DevfileComponentSpec struct {
-	Name       string `json:"componentName,omitempty"`
-	App        string `json:"app,omitempty"`
-	Type       string `json:"type,omitempty"`
-	SourceType string `json:"sourceType,omitempty"`
+	S2IComponents     []Component `json:"s2iComponents"`
+	DevfileComponents []Component `json:"devfileComponents"`
```

Comment:
Thinking out loud. Can we have `Items` and make `S2IComponents` & `DevfileComponents` a part of it? I'm asking this because it'll help us have uniformity with the way k8s returns a list of items. This should also cascade to the way we return list of services, AFAIK.

Reply:
I am thinking as part of refactor we might consolidate these two list with a new label called `IsLegacy` once devfile matures further.

Reply:
So you mean that we can have just one list called `Items` and then in the individual items, there could be a bool label `IsLegacy`  which indicates that it's an s2i component, right? Sounds good to me. Or we could label it `IsS2i`. But that's a separate discussion for when we refactor it.

Reply:
yeah